metashare/api/jobs.py

```diff
@@ -293,3 +297,33 @@ def refresh_github_repositories_for_user(user):
 
 
 refresh_github_repositories_for_user_job = job(refresh_github_repositories_for_user)
+
+
+# This avoids partially-applied saving:
+@transaction.atomic
+def refresh_commits(*, repository, branch_name):
+    """
+    This should only run when we're notified of a force-commit. It's the
+    nuclear option.
+    """
+    from .models import Task
+
+    user = repository.get_a_matching_user()
```

Comment:
This is okay for now, but it's also possible to authenticate with GitHub _as the app_ rather than as the app acting on behalf of a human user, as long as the app has been installed for the repository. that's what this code in CumulusCI handles: https://github.com/SFDO-Tooling/CumulusCI/blob/master/cumulusci/core/github.py#L36

Reply:
Yeah, I'm conscious of that, but deferred because I'd rather get this merged at least now, and that seemed like something that could be done in a later pass.